src/pgduckdb_background_worker.cpp

```diff
@@ -773,6 +784,27 @@ CreateTable(const char *postgres_schema_name, const char *table_name, const char
 		return false;
 	}
 
+	// Record the dependency on the MotherDuck server
+	{
+		HeapTuple new_table_tuple =
+		    SearchSysCache2(RELNAMENSP, CStringGetDatum(table_name), ObjectIdGetDatum(schema_oid));
+		if (HeapTupleIsValid(new_table_tuple)) {
+			Form_pg_class postgres_relation = (Form_pg_class)GETSTRUCT(new_table_tuple);
+			auto oid = postgres_relation->oid;
+			ReleaseSysCache(new_table_tuple);
+			ObjectAddress object_address = {
+			    .classId = RelationRelationId,
+			    .objectId = oid,
+			    .objectSubId = 0,
+			};
+			recordDependencyOnMDServer(&object_address);
```

Comment:
This should be done in our DDL hook instead. That way the dependency is also recorded when someone creates a motherduck table from Postgres, not just when a table is synced to by the background worker.